src/advanced_inv.cpp

```diff
@@ -1037,41 +1071,71 @@ void advanced_inventory::redraw_pane( side p )
     wrefresh( w );
 }
 
-bool advanced_inventory::move_all_items()
+// be explicit with the values
+enum aim_entry {
+    ENTRY_START     = 0,
+    ENTRY_VEHICLE   = 1,
+    ENTRY_MAP       = 2,
+    ENTRY_RESET     = 3
+};
+
+bool advanced_inventory::move_all_items(bool nested_call)
 {
     auto &spane = panes[src];
     auto &dpane = panes[dest];
 
     // AIM_ALL source area routine
     if(spane.get_area() == AIM_ALL) {
-        // future feature :-)
-        popup(_("You can't do that (yet!)"));
-        return false;
-        // make a copy of the current pane for below loop
+        bool done = false;
+        // copy the current pane, to be restored after the move is queued
         auto shadow = panes[src];
         // here we recursively call this function with each area in order to 
         // put all items in the proper destination area, with minimal fuss
         auto &loc = uistate.adv_inv_aim_all_location;
-        // vehicle items for said square
-        if(squares[loc].can_store_in_vehicle()) {
-            // either do the inverse of the pane (if it is the one we are transferring to),
-            // or just transfer the contents (if it is not the one we are transferring to)
-            spane.set_area(squares[loc], (dpane.get_area() == loc) ? !dpane.in_vehicle() : true);
-            // add items, calculate weights and volumes... the fun stuff
-            recalc_pane(src);
-            // then move the items to the destination area
-            move_all_items();
-        }
-        // same as above, but for map items
-        spane.set_area(squares[loc++], false);
-        recalc_pane(src);
-        move_all_items();
-        if(loc > AIM_NORTHEAST) {
-            loc = 0;
+        // re-entry nonsense
+        auto &entry = uistate.adv_inv_re_enter_move_all;
+        // if we are just starting out, set entry to initial value
+        switch(static_cast<aim_entry>(entry++)) {
+            case ENTRY_START:
+                ++entry;
+            case ENTRY_VEHICLE:
+                if(squares[loc].can_store_in_vehicle()) {
+                    // either do the inverse of the pane (if it is the one we are transferring to),
+                    // or just transfer the contents (if it is not the one we are transferring to)
+                    spane.set_area(squares[loc], (dpane.get_area() == loc) ? !dpane.in_vehicle() : true);
+                    // add items, calculate weights and volumes... the fun stuff
+                    recalc_pane(src);
+                    // then move the items to the destination area
+                    move_all_items(true);
+                }
+                break;
+            case ENTRY_MAP:
+                spane.set_area(squares[loc++], false);
```

Comment:
There is a (probably preexisting) bug with the move-all function. If the character stands in the middle of items (spawn a few zombies and kill them to get their loot), select ALL in one pane and a specific location as target (e.g. NW), select move all. The items are moved. Now open the AIM again and repeat the process. Nothing should happen because all items are already at the target. However, the messages indicate items get moved and time passes, this must not happen. It appears to work fine with items in a vehicle cargo trunk.

I think it's this line that unconditionally set the source pane, even if the source pane would be the same as the target pane.